test/generators/home_controller_generator_test.rb

```diff
@@ -17,9 +17,19 @@ class HomeControllerGeneratorTest < Rails::Generators::TestCase
     provide_existing_application_controller
   end
 
-  test "creates the home controller" do
+  test "creates the default home controller with home index view" do
     run_generator
-    assert_file "app/controllers/home_controller.rb"
+
+    assert_file "app/controllers/home_controller.rb", /HomeController < AuthenticatedController/
+    assert_file "app/views/home/index.html.erb"
+    assert_no_file "app/controllers/products_controller.rb"
+  end
+
+  test "creates unauthenticated home controller with home index view given --with_session_token option" do
+    run_generator %w(--with-session-token)
+
+    assert_file "app/controllers/home_controller.rb", /HomeController < ApplicationController/
+    assert_file "app/views/home/index.html.erb"
```

Comment:
Don't you also need to assert that the `products_controller` file was created here?

Reply:
`run_generator` in the context of tests fails to run `generate("anything")` for any other generator called in `ShopifyApp::Generators::HomeControllerGenerator`. 

The reason this isn't an issue is because `generate("shopify_app:products_controller")` is currently tested in `test/generators/products_controller_generator_test.rb`.